src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function add(...args) {
+    if (args.length === 0) {
+      const res = sum;
+
+      sum = 0;
+
+      return res;
+    }
+
+    sum += args.reduce((acc, val) => acc + val, 0);
```

Comment:
Since the `add` function should only receive one argument at a time, there is no need to use the `reduce` method. You can add the single argument directly to the `sum` variable.